commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/CommitStatusPublisherListener.java

```diff
@@ -192,8 +191,8 @@ private void logStatusNotPublished(@NotNull Event event, @NotNull String buildDe
 
   private void runForEveryPublisher(@NotNull Event event, @NotNull SBuildType buildType, @NotNull SBuild build, @NotNull PublishTask task) {
     final String description = LogUtil.describe(build);
-    Set<BuildRevision> publishedRevisions = new HashSet<BuildRevision>();
-    runForEveryPublisher(event, buildType, build, task, description, publishedRevisions);
+    DoubleKeyHashSet<String, BuildRevision> publishedRevisionsByPublisher = new DoubleKeyHashSet<String, BuildRevision>();
```

Comment:
Why not to use Set<Pair<String, BuildRevision>> instead? It would spare you of defining quite a generic private class for quite a specific purpose here.

Reply:
Ah, sorry, was not aware of that class